WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -0,0 +1,120 @@
+import Foundation
+
+import protocol Yosemite.POSItem
+import protocol Yosemite.POSItemProvider
+
+protocol PointOfSaleAggregateModelProtocol {
+    @available(*, deprecated, message: "`allItems` is due for removal, use `itemListState` instead.")
+    var allItems: [POSItem] { get }
+    var itemListState: ItemListState { get }
+
+    func loadInitialItems() async
+    func loadNextItems() async
+    func reload() async
+}
+
+class PointOfSaleAggregateModel: ObservableObject, PointOfSaleAggregateModelProtocol {
+    @Published private(set) var allItems: [POSItem] = []
+    @Published private(set) var itemListState: ItemListState = .initialLoading
+
+    private let itemProvider: POSItemProvider
+
+    private var currentPage: Int = Constants.initialPage
+
+    init(itemProvider: POSItemProvider) {
+        self.itemProvider = itemProvider
+    }
+}
+
+// MARK: - ItemList
+extension PointOfSaleAggregateModel {
+    @MainActor
+    func loadInitialItems() async {
+        itemListState = .initialLoading
+        try? await load(pageNumber: Constants.initialPage)
+    }
+
+    @MainActor
+    func loadNextItems() async {
+        do {
+            itemListState = .loading(allItems)
+            // TODO: Optimize API calls. gh-14186
+            // If there are no more pages to fetch, we can avoid the next call.
+            let nextPage = currentPage + 1
+            try await load(pageNumber: nextPage)
+            currentPage = nextPage
+        } catch {
+            // No need to do anything; this avoids us incorrectly incrementing currentPage.
+        }
+    }
+
+    @MainActor
+    func reload() async {
+        allItems.removeAll()
+        currentPage = Constants.initialPage
+        itemListState = .loading(allItems)
+        try? await load(pageNumber: currentPage)
+    }
+
+    @MainActor
+    private func load(pageNumber: Int) async throws {
```

Comment:
Yep! We use it in `loadNextItems` – we don't care what the error is there, but by having it throw, we can avoid incorrectly incrementing the `currentPage` state. This will help when we replace the full-screen error state for non-initial loads with something less disruptive.